service/src/main/kotlin/fi/espoo/evaka/decision/DecisionService.kt

```diff
@@ -65,136 +64,184 @@ class DecisionService(
-            DecisionType.DAYCARE, DecisionType.PRESCHOOL_DAYCARE, DecisionType.DAYCARE_PART_TIME -> {
-                if (decision.unit.providerType == ProviderType.PRIVATE_SERVICE_VOUCHER) {
-                    listOf(
-                        "daycare/voucher/daycare-decision-page1",
-                        "daycare/voucher/daycare-decision-page2",
-                        "daycare/voucher/daycare-correction",
-                        "daycare/voucher/daycare-acceptance-form"
-                    )
-                } else {
-                    if (application.transferApplication) {
-                        listOf(
-                            "daycare/transfer/daycare-decision-page1",
-                            "daycare/transfer/daycare-decision-page2",
-                            "daycare/transfer/daycare-correction",
-                            "daycare/transfer/daycare-acceptance-form"
-                        )
-                    } else {
-                        listOf(
-                            "daycare/daycare-decision-page1",
-                            "daycare/daycare-decision-page2",
-                            "daycare/daycare-correction",
-                            "daycare/daycare-acceptance-form"
-                        )
-                    }
-                }
-            }
-            DecisionType.PRESCHOOL -> listOf(
-                "preschool/preschool-decision-page1",
-                "preschool/preschool-decision-page2",
-                "preschool/preschool-correction",
-                "preschool/preschool-acceptance-form"
-            )
-            DecisionType.PREPARATORY_EDUCATION -> listOf(
-                "preparatory/preparatory-decision-page1",
-                "preparatory/preparatory-decision-page2",
-                "preparatory/preparatory-correction",
-                "preparatory/preparatory-acceptance-form"
-            )
-        }
-        val isPartTimeDecision: Boolean = decision.type === DecisionType.DAYCARE_PART_TIME
+        return uploadPdfToS3(
+            decisionBucket,
+            constructObjectKey(decision, guardian, decisionLanguage),
+            decisionBytes
+        ).uri
+    }
 
-        // todo: decision templates still except old format :(
-        val legacyApplication = PdfTemplateApplication(
-            form = DaycareFormV0.fromForm2(application.form, application.type, false, false),
-            childDateOfBirth = child.dateOfBirth
+    private fun isDecisionForSecondGuardianRequired(
+        decision: Decision,
+        application: ApplicationDetails,
+        tx: Database.Transaction
+    ) = decision.type != DecisionType.CLUB &&
+        application.otherGuardianId != null &&
+        !personService.personsLiveInTheSameAddress(
+            tx,
+            application.guardianId,
+            application.otherGuardianId
         )
 
-        val pages = templates.mapIndexed { i, template ->
-            Page(
+    companion object {
```

Comment:
Could these be just top level functions or is there some meaning behind putting these inside `companion object`?

Reply:
Yes they could actually, I don't know what was I thinking here.